File: week7/frequency.py
#!/usr/bin/python3

import sys, re, glob

freq = {}
word = sys.argv[1].lower()
for fil in sorted(glob.glob("poems/*.txt")) :
    artist = re.sub(r'poems\/([^\.]*).txt',r'\1',fil)
    artist = ' '.join(artist.split('_'))

    F = open(fil,"r")
    arr = list(filter(None, re.split(r'[^a-zA-Z]+',' '.join(F.readlines()))))
    allCount = len(arr)

    count = len(list(filter(lambda x: x == word,list(map(lambda x: x.lower(), arr)))))

    freq[artist] = count / allCount
    print("%4d/%6d = %.9f %s" % (count, allCount, freq[artist], artist) )
    F.close()

File: week7/frequency.pl
#!/usr/bin/perl

use warnings;
use strict;

my %freq;
my $word = lc $ARGV[0];
foreach my $file (glob "poems/*.txt") {
  my $artist = $file;
  $artist =~ s/poems\/([^\.]*).txt/$1/;
  $artist = join(' ', split(/_/,$artist));

  open(F,"<$file") or die "$0: can't open $file\n";
  my @arr = <F>;
  @arr = split(/[^a-zA-Z]+/, join(' ',@arr));
  @arr = grep { $_ ne '' } @arr;
  my $allCount = scalar @arr;

  my $count = 0;
  foreach my $elem (@arr) {
    chomp $elem;
    $elem = lc $elem;
    if ( $elem eq $word ) {
      $count ++;
    }
  }

  $freq{"$artist"} = $count / $allCount;
  printf("%4d/%6d = %.9f %s\n", $count, $allCount, $freq{"$artist"}, $artist );

}
